code/glTFAsset.h

```diff
@@ -786,8 +790,10 @@ namespace glTF
         void AttachToDocument(Document& doc);
         void DetachFromDocument();
 
+#ifdef  GLTF_ASSET_WITH_EXPORT_SUPPORT
```

Comment:
This is not correct. GCC 6 with LTO gives me this:
warning: virtual table of type ‘struct LazyDict’ violates one definition rule [-Wodr]
note: the conflicting type defined in another translation unit has virtual table table with more entries

Looks like glTFAssetWriter.h defines GLTF_ASSET_WITH_EXPORT_SUPPORT and then includes this header while other translation units don't define it. This violates the one definition rule. I suspect this would break devirtualization in LTO and might even break virtual dispatch in normal compiles.